Assets/Mirror/Components/NetworkTransform/NetworkTransformHybrid2022.cs

```diff
@@ -0,0 +1,1123 @@
+            }
+ #pragma warning restore CS0618
+        }
+
+        // snapshot functions //////////////////////////////////////////////////
+        // construct a snapshot of the current state
+        // => internal for testing
+        protected virtual TransformSnapshot ConstructSnapshot()
+        {
+            // perf
+            target.GetLocalPositionAndRotation(out Vector3 localPosition, out Quaternion localRotation);
+
+            // NetworkTime.localTime for double precision until Unity has it too
+            return new TransformSnapshot(
+                // our local time is what the other end uses as remote time
+                Time.timeAsDouble,
+                // the other end fills out local time itself
+                0,
+                localPosition, // target.localPosition,
+                localRotation, // target.localRotation,
+                Vector3.zero   // target.localScale
+            );
+        }
+
+        // apply a snapshot to the Transform.
+        // -> start, end, interpolated are all passed in caes they are needed
+        // -> a regular game would apply the 'interpolated' snapshot
+        // -> a board game might want to jump to 'goal' directly
+        // (it's easier to always interpolate and then apply selectively,
+        //  instead of manually interpolating x, y, z, ... depending on flags)
+        // => internal for testing
+        //
+        // NOTE: stuck detection is unnecessary here.
+        //       we always set transform.position anyway, we can't get stuck.
+        protected virtual void ApplySnapshot(TransformSnapshot interpolated)
+        {
+            // local position/rotation for VR support
+            //
+            // if syncPosition/Rotation/Scale is disabled then we received nulls
+            // -> current position/rotation/scale would've been added as snapshot
+            // -> we still interpolated
+            // -> but simply don't apply it. if the user doesn't want to sync
+            //    scale, then we should not touch scale etc.
+            if (syncPosition) target.localPosition = interpolated.position;
+            if (syncRotation) target.localRotation = interpolated.rotation;
+            // if (syncScale)    target.localScale    = interpolated.scale;
+        }
+
+        // check if position / rotation / scale changed since last _full reliable_ sync.
+        // squared comparisons for performance
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        bool Changed(Vector3 currentPosition, Quaternion currentRotation)//, Vector3 currentScale)
+        {
+            if (syncPosition)
+            {
+                float positionDelta = Vector3.Distance(currentPosition, lastSerializedBaselinePosition);
+                if (positionDelta >= positionSensitivity)
```

Comment:
You need to have a flag indicating whether at least one delta has been sent since last full reliable sync, in case the thing moved away and back to original pos between full sync intervals. Reset that flag when full sync is sent. if it's true, we need to send one more full sync to move the thing from the last delta that was likely received.